File: samples_c/s72_global_log_level.c
#include <stdio.h>

/* Girdi: tek sayi (0-3 arasi log seviyesi).
   Cikti: seviyeye gore filtrelenmis log satirlari.
   NOT: Kok Neden E (guvensiz global durum) icin 5. bagimsiz ornek. Global
   log seviyesi (g_log_level), set_log_level() ile degistirilir ve
   log_msg() tarafindan okunur - iki fonksiyon arasinda paylasilan
   degistirilebilir durum. */
static int g_log_level = 1; /* 0=DEBUG,1=INFO,2=WARN,3=ERROR */

void set_log_level(int lvl) {
    g_log_level = lvl;
}

void log_msg(int lvl, const char *msg) {
    static const char *names[] = {"DEBUG", "INFO", "WARN", "ERROR"};
    if (lvl >= g_log_level) {
        printf("[%s] %s\n", names[lvl], msg);
    }
}

int main(void) {
    int lvl;
    if (scanf("%d", &lvl) != 1) return 0;
    set_log_level(lvl);
    log_msg(0, "baslangic");
    log_msg(1, "bilgi mesaji");
    log_msg(2, "uyari mesaji");
    log_msg(3, "hata mesaji");
    return 0;
}

File: translations_rust_levelC/s72_global_log_level.rs
use std::io::{self, Read};

static mut G_LOG_LEVEL: i32 = 1;

fn set_log_level(lvl: i32) {
    unsafe {
        G_LOG_LEVEL = lvl;
    }
}

fn log_msg(lvl: i32, msg: &str) {
    let names = ["DEBUG", "INFO", "WARN", "ERROR"];
    unsafe {
        if lvl >= G_LOG_LEVEL {
            println!("[{}] {}", names[lvl as usize], msg);
        }
    }
}

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let lvl: i32 = input.trim().parse().unwrap();
    set_log_level(lvl);
    log_msg(0, "baslangic");
    log_msg(1, "bilgi mesaji");
    log_msg(2, "uyari mesaji");
    log_msg(3, "hata mesaji");
}
